ts/utils/format_number.ts

```diff
@@ -0,0 +1,364 @@
+        // tslint:disable-next-line: no-unnecessary-type-assertion
+        useValue = (value as any)._hex;
+      }
+      newBigNumber = new BigNumber(`${useValue}`, base);
+    } catch (e) {
+        // tslint:disable-next-line: no-console
+        console.error('Error instantiating WrappedBigNumber', e);
+        throw new Error(e);
+    }
+
+    return newBigNumber;
+  };
+
+export const ZERO = createBigNumber(0);
+export const TEN = createBigNumber(10, 10);
+
+export const ETHER_NUMBER_OF_DECIMALS = 5;
+export const ZRX_NUMBER_OF_DECIMALS = 2;
+
+export const SMALLEST_NUMBER_DECIMAL_PLACES = 8;
+export const USUAL_NUMBER_DECIMAL_PLACES = 4;
+
+export interface FormattedNumber {
+  fullPrecision: number | string;
+  roundedValue: number | BigNumber;
+  roundedFormatted: string;
+  formatted: string;
+  formattedValue: number | string;
+  denomination: string;
+  minimized: string;
+  value: number;
+  rounded: number | string;
+  full: number | string;
+}
+
+export interface FormattedNumberOptions {
+  decimals?: number;
+  decimalsRounded?: number;
+  denomination?: (value: string) => string;
+  roundUp?: boolean;
+  roundDown?: boolean;
+  positiveSign?: boolean;
+  zeroStyled?: boolean;
+  minimized?: boolean;
+  blankZero?: boolean;
+  bigUnitPostfix?: boolean;
+  removeComma?: boolean;
+}
+
+function formatNone(): FormattedNumber {
+  return {
+    value: 0,
+    formattedValue: 0,
+    formatted: '-',
+    roundedValue: 0,
+    rounded: '-',
+    roundedFormatted: '-',
+    minimized: '-',
+    denomination: '',
+    full: '-',
+    fullPrecision: '0',
+  };
+}
+
+function formatBlank(): FormattedNumber {
+  return {
+    value: 0,
+    formattedValue: 0,
+    formatted: '',
+    roundedValue: 0,
+    rounded: '',
+    roundedFormatted: '',
+    minimized: '',
+    denomination: '',
+    full: '',
+    fullPrecision: '0',
+  };
+}
+
+function optionsBlank(): FormattedNumberOptions {
+  return {
+    decimals: 0,
+    decimalsRounded: 0,
+    denomination: (v => ''),
+    roundUp: false,
+    roundDown: false,
+    positiveSign: false,
+    zeroStyled: true,
+    minimized: false,
+    blankZero: false,
+    bigUnitPostfix: false,
+  };
+}
+
+// tslint:disable-next-line: typedef
+function addBigUnitPostfix(value: BigNumber, formattedValue: string | number, removeComma: boolean = false) {
+  let postfixed;
+  if (value.gt(createBigNumber('1000000000000', 10))) {
+    postfixed = '> 1T';
+  } else if (value.gt(createBigNumber('10000000000', 10))) {
+    postfixed =
+      value.dividedBy(createBigNumber('1000000000', 10)).toFixed(0) + 'B';
+  } else if (value.gt(createBigNumber('10000000', 10))) {
+    postfixed =
+      // tslint:disable-next-line: prefer-template
+      value.dividedBy(createBigNumber('1000000', 10)).toFixed(0) + 'M';
+  } else if (value.gt(createBigNumber('10000', 10))) {
+    postfixed = value.dividedBy(createBigNumber('1000', 10)).toFixed(0) + 'K';
```

Comment:
NIT: I would argue that `1e12`, `1e9` etc is slightly more readable since you don't have to count the 0s